node/node.go

```diff
@@ -985,9 +1027,31 @@ func (n *Node) PrivValidator() types.PrivValidator {
 	return n.privValidator
 }
 
-// GenesisDoc returns the Node's GenesisDoc.
-func (n *Node) GenesisDoc() *types.GenesisDoc {
-	return n.genesisDoc
+// GenesisDoc returns a GenesisDoc object after reading the genesis file from disk.
+// The function does not check for the genesis's validity since it was already
+// checked at startup, and we work under the assumption that correct nodes (i.e.,
+// non-Byzantine) are not compromised. Therefore, their file system can be
+// trusted while the node is running.
+// Note that the genesis file can be large (hundreds of MBs, even GBs); therefore,
+// we recommend that the caller does not keep the GenesisDoc returned by this
+// function in memory longer than necessary.
+func (n *Node) GenesisDoc() (*types.GenesisDoc, error) {
```

Comment:
Not in our codebase. Tbh, my first instinct was to delete it. However, because this is an exported method of an exported type, there might be users relying on it. Therefore, I left it there, but changed its signature to adapt it to the changes.

It's still a breaking change, but less breaking than removing the method altogether.

Reply:
sounds good, thanks for the clarification